xrpl/transaction/main.py

```diff
@@ -29,25 +35,26 @@ def safe_sign_and_submit_transaction(
     Returns:
         The response from the ledger.
     """
-    tx_blob = safe_sign_transaction(transaction, wallet)
+    tx_blob = safe_sign_transaction(transaction, wallet, client)
     return submit_transaction_blob(tx_blob, client)
 
 
-def safe_sign_transaction(transaction: Transaction, wallet: Wallet) -> str:
+def safe_sign_transaction(
+    transaction: Transaction, wallet: Wallet, client: Client
+) -> str:
```

Comment:
I'm not sure it's a good idea to have a client variable in `sign` when it's supposed to be local (seems to be a little bit of mixed messages), the only other idea I have is an optional variable `should_autofill` that defaults to True, and if False, doesn't do any autofill (and then you don't need to provide a Client either).

Reply:
See my discussion of the preferred transaction signing and submission flow (  #189 ).

Autofill functionality should move to a "prepare" step while signing remains completely offline. (Optionally there can be an online method that combines "prepare" and "sign" too.)